src/cdomains/apron/sharedFunctions.apron.ml

```diff
@@ -6,7 +6,8 @@ open GobApron
 
 module M = Messages
 
-let int_of_scalar ?round (scalar: Scalar.t) =
+
+let int_of_scalar ?(scalewith=Z.one) ?round (scalar: Scalar.t) =
```

Comment:
Now looking at it again, it seems odd that `int_of_scalar` would have the `scalewith` argument. All it does is do an extra multiplication at the very end. That multiplication could just be done by whoever calls `int_of_scalar` themselves without overloading this function with even more complexity.

I would understand that need if the scaling happened before the rounding division or something, but it even doesn't.

Reply:
The rounding division part is not the crucial point here, the scaling is only done there for consistency. The main focus is the part, where integer division is done - here, the scaling happens before the division. This is also, why we can not pull this out of int_of_scalar imho

Reply:
Right, nevermind. I thought for a moment that if it's exact division, then it shouldn't matter, but it does when the factors of `d` are split between `scale` and `n`.